nomad/leader.go

```diff
@@ -1230,3 +1236,24 @@ func (s *Server) getOrCreateAutopilotConfig() *structs.AutopilotConfig {
 
 	return config
 }
+
+// getOrCreateSchedulerConfig is used to get the scheduler config, initializing it if necessary
```

Comment:
Expand the comment a little and say in a fresh cluster, there won't be a  scheduler config so we need to initialize it